File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.java
class Solution {
    public int maxCoins(int[] piles) {
        int ans = 0;
		Arrays.sort(piles);
		
		int counter = piles.length / 3, indx = piles.length - 2;
		
		do
		{
			ans += piles[indx];
			indx -= 2;
			counter--;
		} while (counter > 0);
		
		return ans;
    }
}

File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.py
class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        piles.sort()
        max_sum = 0
        count = i = -2
        while i >= -len(piles) and i >= (count * len(piles) // 3):
            max_sum += piles[i]
            i -= 2
        return max_sum